src/functions.sh

```diff
@@ -334,8 +334,15 @@ get_cake_lla_string() {
 sqm_start_default() {
     #sqm_error "sqm_start_default"
     [ -n "$IFACE" ] || return 1
-    #do_modules
-    #verify_qdisc $QDISC "cake" || return 1
+    
+    fn_exists sqm_prepare_script
+    if [ "$?" -eq "0" ]; then
+        sqm_prepare_script
+    fi
+    
+    
+    do_modules
```

Comment:
So all the scripts will now call do_modules twice? Once here, and once in their prepare() functions?

Reply:
Yes, otherwise we have hidden dependencies as the prepare function is optional.

Reply:
Well, I meant you should remove them from the scripts, not the common function :)

Reply:
I believe that is not an option currently. sqm_start_default() only checks the bare minimum ($QDISC as it does not know any better, nor should it) but needs to be certain modules are loaded.
The qos scripts however, need to heck spcific things like "verify_qdisc "htb" || return 1" for simple.qos, as far as I can tell that also requires modules to be loaded. 
I do believe that we fixed do_modules to do nothing if modules are already loaded, so calling it twice should have no side-effects. The alternative would be to call this from start_sqm.sh, then we are certain modules are loaded and we can avoid doing it deeper in the scripts?

Reply:
You could just reorder things so do_modules is before the sqm_prepare_script call?

Reply:
For the current scripts that would work, but anybody overriding sqm_start_default() will need to do this themselves without any pointers anywhere... So maybe rally move this to start_sqm.sh instead and remove all lower instances?

Reply:
Presumably someone overriding sqm_start_default() will have to know to replicate its functionality anyway, or things will break in other interesting ways? :)